src/polars_dataframe.rs

```diff
@@ -0,0 +1,71 @@
+use polars_core::{
+    prelude::{DataFrame, Series},
+    utils::{
+        accumulate_dataframes_vertical_unchecked,
+        rayon::prelude::{IndexedParallelIterator, IntoParallelIterator, ParallelIterator},
+    },
+};
+
+use super::{arrow::datatypes::SchemaRef, Statement};
+
+/// An handle for the resulting Polars DataFrame of a query.
+#[must_use = "Polars is lazy and will do nothing unless consumed"]
+pub struct Polars<'stmt> {
+    pub(crate) stmt: Option<&'stmt Statement<'stmt>>,
+}
+
+impl<'stmt> Polars<'stmt> {
+    #[inline]
+    pub(crate) fn new(stmt: &'stmt Statement<'stmt>) -> Polars<'stmt> {
+        Polars { stmt: Some(stmt) }
+    }
+
+    /// return arrow schema
+    #[inline]
+    pub fn get_schema(&self) -> SchemaRef {
+        self.stmt.unwrap().stmt.schema()
+    }
+}
+
+impl<'stmt> Iterator for Polars<'stmt> {
+    type Item = DataFrame;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let column_names = self.stmt.unwrap().column_names();
+        let arrow_struct_array = self.stmt?.step()?;
+        let (_, arrow_arrays, _) = arrow_struct_array.into_parts();
+        let series_vec = arrow_arrays
+            .into_par_iter()
+            .enumerate()
+            .map(|(i, arrow_array)| {
+                let arrow_array_data = arrow_array.to_data();
+                let arrow2_array = arrow2::array::from_data(&arrow_array_data);
+                let name = column_names[i].as_str();
+
+                Series::try_from((name, arrow2_array)).unwrap()
+            })
+            .collect::<Vec<_>>();
+
+        Some(DataFrame::new_no_checks(series_vec))
+    }
+}
+
+/// Polars DataFrame wrapper
```

Comment:
Why we need the wrapper?

Reply:
The result of `query_arrow` is `Iterator of RecordBatch`. 
I want to keep this pattern, so similarly, the result of 'query_polars' is `Iterator of DataFrame` ( 1 RecordBatch is corresponding to 1 DataFrame). Afterall, we need a method to convert `Iterator of DataFrame` into single one `DataFrame`.

It is not possible to `impl FromIterator<DataFrame> for DataFrame` directly. Therefore, there're possibly 4 ways to do this:

1. Creating a wrapper of polars DataFrame like I did.
Usage:
```rust
let df = stmt.query_polars([])?.collect::<PolarsDataFrame>().into_inner();
```

2.  Creating a trait like 'IntoPolarsDataFrame'. 
```rust
trait IntoPolarsDataFrame {
    fn into_polars_dataframe(self) -> DataFrame;
}

impl<T> IntoPolarsDataFrame for T
where
    T: IntoIterator<Item = DataFrame>,
{
    fn into_polars_dataframe(self) -> DataFrame {
        accumulate_dataframes_vertical_unchecked(self)
    }
}
```
Usage:
```rust
let df = stmt.query_polars([])?.into_polars_dataframe();
```
3. Use a function:
```rust
fn into_polars_dataframe<T:    T: IntoIterator<Item = DataFrame>(T) -> DataFrame {
  accumulate_dataframes_vertical_unchecked(self)
}
```
Usage:
```rust
let df = into_polars_dataframe(stmt.query_polars([])?);
```
4. Make `query_polars` return a single DataFrame directly, `Iterator of DataFrame`.

Currently, I think option 2 looks the best

Reply:
Yes I already read this, I mean we have impl the iterator for Polars, and can collect DataFrame from this, so we don't need this wrapper

Reply:
You can read `arrow_batch.rs` file again

Reply:
We can `collect::<Vec<DataFrame>>()` but I believe we should offer a method to convert this `Vec<DataFrame>` into a single `DataFrame`. Therefore, I proposed 4 options as above.

Reply:
I suggest we don't do this in this crate, user have many ways to combine a Vec.

And when duckdb returns a Vec of DataFrame, it means the result set is big enough and we should steam the data, convert into one might consume a lot of memory and is a bad practise.

Reply:
That quite makes sense. Converting `Vec<DataFrame>` to `DataFrame` may be beyond this crate's scope. 

In my case, it was not easy to find out the method `accumulate_dataframes_vertical_unchecked`.

Therefore, my thought is that I will the remove the wrapper code but can I put some suggestions about converting `Vec<DataFrame>` to `DataFrame`  in the docstring of the `query_polars` function? I believe this will be helpful for users.